src/data_write.py

```diff
@@ -8,175 +8,213 @@
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    experiment_name: str = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    first_range: int = field(
+        metadata={'groups': ['bfiq', 'rawacf']})
+    first_range_rtt: float = field(
+        metadata={'groups': ['bfiq', 'rawacf']})
+    freq: float = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf']})
+    gps_locked: bool = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    gps_to_system_time_diff: float = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    int_time: float = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    intf_acfs: np.ndarray = field(
+        metadata={'groups': ['rawacf']})
+    intf_antenna_count: int = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    lags: np.ndarray = field(
+        metadata={'groups': ['bfiq', 'rawacf']})        # TODO: Should this be in antennas_iq too? Or removed from bfiq?
+    lp_status_word: int = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    main_acfs: np.ndarray = field(
+        metadata={'groups': ['rawacf']})
+    main_antenna_count: int = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    noise_at_freq: np.ndarray = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf']})
+    num_ranges: int = field(
+        metadata={'groups': ['bfiq']})                              # TODO: Does this need to be in more file types?
+    num_samps: int = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawrf']})
+    num_sequences: int = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    num_slices: int = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    pulse_phase_offset: np.ndarray = field(
+        metadata={'groups': ['antennas_iq', 'bfiq']})
+    pulse_sample_start: List[float] = field(
+        metadata={'groups': ['txdata']})
+    pulse_timing_us: List[float] = field(
+        metadata={'groups': ['txdata']})
+    pulses: np.ndarray = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf']})
+    range_sep: int = field(
+        metadata={'groups': ['bfiq', 'rawacf']})                      # TODO: Does this need to be in antennas_iq?
+    rx_center_freq: float = field(
+        metadata={'groups': 'rawrf'})
+    rx_sample_rate: float = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    samples_data_type: str = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    scan_start_marker: bool = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
+    scheduling_mode: str = field(
+        metadata={'groups': ['antennas_iq', 'bfiq', 'rawacf', 'rawrf']})
```

Comment:
Looking at all these parameters that are repeated across file types has got me thinking: it may be a good idea to change they way we are documenting our different data files within ReadTheDocs. Right now we have separate files for each type, but since there are so many similarities, combining all data types into one documentation page would clean things up a fair bit. 

We could at least combine the parameter tables into a single one - this would show the differences between the data file types quite well. 

https://github.com/SuperDARNCanada/borealis/blob/develop/docs/source/antennas_iq.rst

Reply:
I think that could be useful for us as developers, but I'm not sure that it would be useful to end users of our data. In my opinion, the pages as they are now are great because the users can refer to it to understand the structure of whatever type of data file they have, and it isn't muddled with fields from other types of data files. Not all fields make sense at all levels of data either, for example `pulse_phase_offset` is used to generate rawacf files but is not required to understand the resulting rawacf file.